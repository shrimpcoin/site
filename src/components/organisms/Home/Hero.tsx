import React from 'react';
import styled from 'styled-components';

import { Button } from 'components/atoms/Button';

import { ReactComponent as Shrimpcoin } from 'assets/images/shrimpcoin.svg';
import { ReactComponent as Wave } from 'assets/images/wave.svg';

const Container = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  background: white;
`;

const Content = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 20vh;
`;

const Text = styled.div`
  width: 20%;
  margin-left: 10%;
  font-size: 2.5em;
  font-family: Manrope;
  font-weight: 200;
`;

export const Hero: React.FC = (): React.ReactElement => (
  <Container>
    <Content>
      <Shrimpcoin />
      <Text>
        <p>It's time to get shrimp done.</p>
        <Button background="var(--color-primary)" outline="white" href="/wallet">
          Get Started
        </Button>
      </Text>
    </Content>
    <Wave style={{ position: 'absolute', bottom: '0%', width: '100%', height: '20%' }} preserveAspectRatio="none" />
  </Container>
);
