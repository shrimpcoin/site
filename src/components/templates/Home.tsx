import React from 'react';
import styled from 'styled-components';

import { Header } from 'components/molecules/Header';
import { Hero } from 'components/organisms/Home/Hero';
import { One as FAQOne } from 'components/organisms/FAQ/One';
import { Two as FAQTwo } from 'components/organisms/FAQ/Two';
import { Three as FAQThree } from 'components/organisms/FAQ/Three';
import { Four as FAQFour } from 'components/organisms/FAQ/Four';
import { Five as FAQFive } from 'components/organisms/FAQ/Five';
import { Footer } from 'components/molecules/Footer';

const Container = styled.div`
  position: relative;
  height: 390vh;
`;

const FAQGrid = styled.div`
  display: grid;
  grid-template-rows: 80vh 30vh 40vh 60vh 100vh 80vh;
  position: absolute;
  top: 0px;
  left: 0px;
  width: 100%;
`;

export const Home: React.FC = (): React.ReactElement => (
  <Container>
    <Header scrollDist={300} />
    <FAQGrid>
      <Hero />
      <FAQOne />
      <FAQTwo />
      <FAQThree />
      <FAQFour />
      <FAQFive />
    </FAQGrid>
    <Footer />
  </Container>
);
