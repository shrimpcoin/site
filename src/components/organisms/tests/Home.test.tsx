import React from 'react';
import renderer from 'react-test-renderer';
import { Hero } from 'components/organisms/Home/Hero';

test('Hero', () => {
  const component = renderer.create(<Hero />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
