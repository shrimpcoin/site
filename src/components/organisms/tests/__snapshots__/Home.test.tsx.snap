// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero 1`] = `
<div
  className="sc-gtsrHT iazTCh"
>
  <div
    className="sc-dlnjwi fouFvp"
  >
    <svg>
      shrimpcoin.svg
    </svg>
    <div
      className="sc-hKFxyN kFKRQu"
    >
      <p>
        It's time to get shrimp done.
      </p>
      <a
        className="sc-bdnxRM dhIpaF"
        href="/wallet"
      >
        Get Started
      </a>
    </div>
  </div>
  <svg
    preserveAspectRatio="none"
    style={
      Object {
        "bottom": "0%",
        "height": "20%",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    wave.svg
  </svg>
</div>
`;
